src/evict/evict_lru.c

```diff
@@ -841,10 +780,15 @@ __wt_evict_file_exclusive_on(WT_SESSION_IMPL *session)
 	 * this point.
 	 */
 	F_SET(btree, WT_BTREE_NO_EVICTION);
+	F_SET(cache, WT_CACHE_CLEAR_WALKS);
 	WT_FULL_BARRIER();
 
 	/* Clear any existing LRU eviction walk for the file. */
-	WT_ERR(__evict_request_clear_walk(session));
+	__wt_spin_lock(session, &cache->evict_pass_lock);
+	ret = __evict_clear_walk(session);
+	__wt_spin_unlock(session, &cache->evict_pass_lock);
```

Comment:
Can you please make a `WT_WITH_EVICT_PASS_LOCK` macro that uses `WT_WITH_LOCK` to acquire `evict_pass_lock` and set a session flag?  We should be able to assert in `__evict_clear_walk` that we are holding the lock and at the moment we can't.